### ISSUE TITLE
Lookup via Archive.org identifier

### ISSUE THREAD

**Issue by KayBeSee** [RelistenNet/RelistenApi#53]
Hey Alec and friends,

I'm working on a project right now that is linking a webpage to Archive.org content. It would be nice to provide a link to open a show in the relisten app.

Looking at the code, I don't see a way to look up a Relisten show URL by its archive.org identifier (i.e. `jrad2022-12-04` => `https://relisten.net/jrad/2022/12/04` or `rre2004-09-08.shnf` => `https://relisten.net/railroad-earth/2004/09/08?source=58709`).

Am I overlooking something? 

Another alternative would be a route to get the Relisten artistIdOrSlug from the upstream_identifier and then manually parse the date, which would be fine too.

**Comment by switz** [comment]
I don’t believe there’s a way to lookup a Relisten source via archive identifier, seems like your proposed solution would be reasonable enough without adding any new routes.

That said, if you want to send a PR with a new API route, I think we’d be happy to merge it.
